hbase_sql/src/main/java/com/splicemachine/derby/lifecycle/HEngineSqlEnv.java

```diff
@@ -169,4 +179,177 @@ public ServiceDiscovery serviceDiscovery() {
     }
 
 
+    /**
+     * Parse a Spark or Hadoop size parameter value, that may use k, m, g or t
+     * to represent kilobytes, megabytes, gigabytes or terabytes, respectively,
+     * and return back the corresponding number of bytes.
+     * @param sizeString the parameter value string to parse
+     * @param defaultValue the default value of the parameter if an invalid
+     *                     <code>sizeString</code> was passed.
+     * @return The value in bytes of <code>sizeString</code>, or <code>defaultValue</code>
+     *         if a <code>sizeString</code> was passed that could not be parsed.
+     */
+    private static long parseSizeString(String sizeString, long defaultValue) {
```

Comment:
i know we don't do a lot of unit tests, but this function would be very easy to unit test, please add a test.

Reply:
I've added unit tests.